### DISCUSSION TITLE
Where does pricing information on a part come from?

### DISCUSSION THREAD

**Discussion by method1020** [inventree/InvenTree#4232] [Q&A]
I have some parts in my Inventree install that were purchased from a supplier with a reference PO in inventree, the PO has the pricing information in it. When does the pricing information on the actual part get updated?

![Screenshot 2023-01-19 120943](https://user-images.githubusercontent.com/3278824/213513283-a0807b8b-11ee-411a-a3be-587635e5b53d.jpg)

Thank you,

**Comment by matmair** [comment]
Generally, it comes from stock data and should be updated immediately. Do you have pricing enabled on the part?

**Comment by method1020** [comment]
Hi @matmair , thank you for the fast response.

Is this a global setting I need to turn on or per part. I am not seeing the option to turn it on. (v0.9.0)

**Reply by matmair** [reply]
@method1020 This is per part and can be changed in the main part view with the `Edit Part` action.

**Reply by method1020** [reply]
I must be missing it.

I navigate to the part, then:

`Part Actions` > `Edit Part`

Navigate to `Part Attributes` and listed are:

- [X]  Component
- [ ] Assembly
- [ ] Is Template
- [ ] Trackable
- [X] Purchasable
- [X] Salable
- [ ] Virtual
- [X] Active

In the menu options above I have:

- Category
- Name
- IPN
- Revision
- Description
- Variant Of
- Keywords
- Units
- Link
- Default Location
- Default Supplier
- Minimum Stock

Is this the section?

**Reply by matmair** [reply]
That would be the correct setting to get sale and purchase prices calculated.
Could you try creating a part and order in the demo and check if it is working there? Sounds more and more like it could be a bug.

**Reply by method1020** [reply]
I re-ran a test with actual data and I am not seeing pricing info. Maybe a bug?

1.  Ordered a supplier part from supplier (10.01.06.03499)
2.  Received the supplier parts with `Receive Line Item` button.
3.  I see the 'stock' part in my part stock. It shows the purchased price next to it.
4. If I click the Pricing tab, I do not see any data.

Looking at top line item...
![Screenshot 2023-01-20 085345](https://user-images.githubusercontent.com/3278824/213714447-a69d2c29-d91e-4737-87ff-6c2a3d27a8ec.jpg)

I see the price...
![Screenshot 2023-01-20 085456](https://user-images.githubusercontent.com/3278824/213714455-1f8e8007-8cfb-4022-a458-227e0e57c48f.jpg)

No pricing information shown here...
![Screenshot 2023-01-20 085518](https://user-images.githubusercontent.com/3278824/213714452-d71e5e52-1ab9-4312-ba13-93836f241bc4.jpg)

**Reply by matmair** [reply]
I tried reproducing this in the demo (it is running 0.9.2 so a tad newer than your version). There everything seems to work fine.

@method1020 Would you mind filing a bug report? We might have to look deeper into this and it easier for the team to track stuff with that template.
https://github.com/inventree/InvenTree/issues/new/choose

**Comment by method1020** [comment]
https://github.com/inventree/InvenTree/issues/4248
